Week04/lesson08/homework/temp_converter/js/index.js

```diff
@@ -0,0 +1,15 @@
+function convert() {
+
+
+  var c = document.querySelector("#celciusInput").value;
+  var f = c * 1.8 + 32
+
+  c = parseInt(c);
+  f = parseInt(f);
+  // I included this, but seems to work without it also. Confused!
```

Comment:
Right, to answer your question:

So, in the same way that you can do:

`4 + ""`

...and get: `"4"` as a string, Javascript is nice sometimes about "figuring out" what you mean. So while sometimes it'll blow up for you trying to do maths with a string (text), other times, like here, it'll let it slide.

Basically, when `c * 1.8 + 32` occurs, JS is being nice and converting `c` into a number for you.

That said, it's nice to be thorough as well as guard against an evil user that might try to put text instead of numbers into that `<input>`, so doing `parseFloat` on `c` is a good idea. I say `parseFloat` and not `parseInt` (which is what you've done) because you need to consider the possibility that the user might put a decimal value in.

Also, you only need to convert `c`, because `c` is the only value you're pulling off the DOM. `f`'s value is determined by `c` and the maths done on `c`, so you don't need to `parse` it in any way.

So basically, you're looking at:

```
var c = document.querySelector("#celciusInput").value;
var f = parseFloat(c) * 1.8 + 32;
```